### DISCUSSION TITLE
Small objects in high resolution pictures (and videos) are not recognized

### DISCUSSION THREAD

**Discussion by Serhiimagg** [ultralytics/ultralytics#3788] [General]
I'm trying to recognize a small object in a high resolution image (or video). I found several similar questions but didn't find a solution. After a lot of testing, I prepared a simple test: an image of a ball. Two pictures with different resolutions, pixel by pixel the balls are exactly the same, the difference is only in the outer frame. Images 640x640 and 1280x1280. So, in the 640x640 picture, the ball is confidently recognized (I also tried it on the video), but in the 1280x1280 picture it is not recognized. 
I am using YOLOV5, Windows 10 Pro OS. I renamed the startup file (detect_crop.py) because I don't want to make changes to the original file. I tried explicitly specifying the image size (--imgsz 1280) but the result is the same
![balls_frame](https://github.com/ultralytics/ultralytics/assets/62691558/303178e8-3f93-42ce-bb63-1109d06b54c4)
![balls_pixel](https://github.com/ultralytics/ultralytics/assets/62691558/f9fb1105-ceab-4718-b4ac-8e683f54da91)
![cmd_1](https://github.com/ultralytics/ultralytics/assets/62691558/572e2ff8-4017-46f6-a814-c45017387538)

**Comment by glenn-jocher** [comment]
@Serhiimagg hello! It seems you're facing an issue with detecting small objects in high-resolution images using YOLOv5. However, since you've reached out on the YOLOv8 repository, I'll guide you based on YOLOv8's capabilities, which may also be applicable to your situation.

Detecting small objects can be challenging due to the reduced feature information available at lower scales. In YOLOv8, there are a few strategies you could consider:

1. **Data Augmentation**: Ensure that your training dataset includes a variety of scales of your object of interest. Data augmentation can help the model generalize better to different sizes.

2. **Model Architecture**: Choose a model architecture that is known to perform well on small objects. Some architectures have features that specifically aid in detecting smaller objects.

3. **Input Resolution**: When training or inferring with YOLOv8, you can experiment with different input resolutions. Higher resolutions can sometimes help with small object detection, but they also require more computational resources.

4. **Fine-tuning**: If you have a pre-trained model, you can fine-tune it on a dataset that is representative of your use case, with many examples of the small objects you're trying to detect.

5. **Anchor Boxes**: Review the anchor box sizes that the model is using. If the predefined anchors are too large for your small objects, the model might struggle to detect them. Customizing anchor boxes to better match the scale of your objects can improve detection.

6. **Post-processing**: Adjust the confidence threshold and non-maximum suppression (NMS) settings during inference to ensure that smaller detections are not filtered out.

For more detailed guidance on these strategies, please refer to our documentation at https://docs.ultralytics.com. If you continue to face issues, consider opening an issue in the YOLOv8 repository with details of your approach, and we'll be happy to assist you further. Remember, the key is to ensure that the model is exposed to a variety of object sizes during training and that the inference settings are tuned to your specific needs. Good luck! 🚀
